bin/build.sh

```diff
@@ -1,73 +1,81 @@
-edudoc_stylesheet=$(realpath "./assets/edudoc-style.css")
+main_stylesheet=$(realpath "./assets/style.css")
 edudoc_header=$(realpath "./assets/edudoc-header.html")
 # Current date
 compile_date=$(date '+%Y-%m-%d')
 
-# Function for converting Markdown files from a given folder into PDF files using the given stylesheet
-# The first argument ($1) is the folder and the second argument is the stylesheet ($2)
+# Function for converting Markdown files from a given directory into PDF files.
+# The first argument ($1) refers to the directory name (e.g: documents or edudoc).
 convert_to_pdf() {
+  rm -rf "build/$1"
   cp -r "$1/" build/
 
-  # Find formal and legal Markdown files and build PDFs out of them.
+  # Find Markdown files and build PDFs out of them.
   find "build/$1" -name '*.md' | while read file; do
-    echo "Processing $file using the ${1} stylesheet"
+    echo "Converting ${file#build/}..."
 
-    pdf_name="${file%.md}.pdf"
-    html_name="${file%.md}.html"
-    
-    file_headline=$(head -n 1 "$file")
-    document_title=$(echo "$file_headline" | sed -E "s/#+\s*//")
+    pdf_path="${file%.md}.pdf"
+    html_path="${file%.md}.html"
+    # Use first header as the document title; remove all # characters and whitespace from the beginning
+    document_title=$(head -n 1 "$file" | sed -E "s/^#+\s*//")
+    # Absolute path to the custom stylesheet
+    custom_stylesheet=$(realpath "./assets/$1-style.css")
 
     # Replace wca{...} and wcadoc{...} with absolute WCA URLs (the correct URLs only get inserted in production)
     sed -Ei "s#wca\{([^}]*)\}#$wca_url\1#g" "$file"
     sed -Ei "s#wcadoc\{([^}]*)\}#$wca_docs_url\1#g" "$file"
     # Replace {logo} with the path to the WCA logo in /assets
     sed -Ei "s#\{logo\}#$logo#g" "$file"
 
-    # This creates the custom header for each edudoc and saves it next to the doc that will use it
-    if [ $1 = "edudoc" ]; then
-      header_html="${file%.md}-header.html"
-    
+    # This creates the custom header for each edudoc and prepends it to each file
+    if [ "$1" = "edudoc" ]; then
       # Set the document title, path to the logo and the date
-      sed -E "s#DOCUMENT_TITLE#$document_title#g" "$edudoc_header" |
-      sed -E "s#WCA_LOGO_PATH#$logo#g" |
-      sed -E "s#DATE#$compile_date#g" > "$header_html"
+      sed -E "s#\{document_title\}#$document_title#g" "$edudoc_header" |
+      sed -E "s#\{wca_logo_path\}#$logo#g" |
+      sed -E "s#\{date\}#$compile_date#g" |
+      echo -e "$(cat /dev/stdin)\n\n$(cat $file)" > "$file"
     fi
 
-    # Markdown -> HTML
-    pandoc -s --from markdown --to html5 --css=$2 --metadata pagetitle="$document_title" "$file" -o "$html_name"
+    # Markdown => HTML + remove unsupported styles generated by pandoc
+    pandoc -s --from markdown --to html5 --css "$main_stylesheet" --css "$custom_stylesheet" --metadata pagetitle="$document_title" "$file" |
+    sed -E "s#gap: min\(4vw, 1\.5em\);##" |
+    sed -E "s#overflow-x: auto;##" > "$html_path"
```

Comment:
What is this `sed` doing?

Reply:
Yeah, so as the comment above it mentions, there is this totally unnecessary styling that pandoc generates that's not supported by weasyprint, and this sed just removes it. I have not found this affecting anything, it just removes the annoying warnings during the build process.